src/tdsops.f90

```diff
@@ -0,0 +1,682 @@
+module m_tdsops
+   use m_common, only: dp, pi
+
+   implicit none
+
+   type :: tdsops_t
+      !! Tridiagonal Solver Operators class.
+      !! Operator arrays are preprocessed in this class based on the arguments
+      !! provided. dist_fw and dist_bw are used in the first phase of the
+      !! distributed tridiagonal solver algorithm. dist_sa and dist_sc are used
+      !! in the final substutition phase. See the kernels_dist.f90 files in the
+      !! relevant backend folders.
+      !! coeff arrays define the specific rules of building the RHS
+      !! corresponding to the tridiagonal system to be solved, and used only in
+      !! the first phase of the distributed algorithm when building the RHS.
+      !! If a boundary condition is defined then coeffs_t and coeffs_b differ
+      !! from coeffs array and define the RHS rule for the first and last 4
+      !! entries in the tridiagonal system (n_halo = 4).
+      !!
+      !! This class does not know about the current rank or its relative
+      !! location among other ranks. All the operator arrays here are used when
+      !! executing a distributed tridiagonal solver phase one or two.
+      real(dp), allocatable, dimension(:) :: dist_fw, dist_bw, dist_sa, dist_sc
+      real(dp), allocatable :: coeffs(:), coeffs_t(:, :), coeffs_b(:, :)
+      integer :: n, n_halo
+      real(dp) :: alpha, delta
+   contains
+      procedure :: deriv_1st, deriv_2nd, interpl_mid, stag_der_1st, preprocess
+   end type tdsops_t
+
+   interface tdsops_t
+      module procedure tdsops
+   end interface tdsops_t
+
+contains
+
+   type(tdsops_t) function tdsops(n, delta, operation, variation, n_halo, &
+                                  from_to, bc_top, bc_bot, paire, c_nu, nu0_nu)
+      implicit none
+
+      integer, intent(in) :: n
+      integer, optional, intent(in) :: n_halo
```

Comment:
It is easier to read when the arguments are in the same order here and in the function description.

I think it is interesting to have optional arguments with a prefix opt_ in their name. Reading the code is easier then.

Reply:
Fixed